UiPath.Ipc.net6.0-windows.received.txt

```diff
@@ -0,0 +1,264 @@
+        bool IsConnected();
+    }
+    public static class IOHelpers
+    {
+        public static System.IO.Pipes.PipeSecurity Allow(this System.IO.Pipes.PipeSecurity pipeSecurity, System.Security.Principal.IdentityReference sid, System.IO.Pipes.PipeAccessRights pipeAccessRights) { }
+        public static System.IO.Pipes.PipeSecurity Allow(this System.IO.Pipes.PipeSecurity pipeSecurity, System.Security.Principal.WellKnownSidType sid, System.IO.Pipes.PipeAccessRights pipeAccessRights) { }
+        public static System.IO.Pipes.PipeSecurity AllowCurrentUser(this System.IO.Pipes.PipeSecurity pipeSecurity, bool onlyNonAdmin = false) { }
+        public static System.IO.Pipes.PipeSecurity Deny(this System.IO.Pipes.PipeSecurity pipeSecurity, System.Security.Principal.IdentityReference sid, System.IO.Pipes.PipeAccessRights pipeAccessRights) { }
+        public static System.IO.Pipes.PipeSecurity Deny(this System.IO.Pipes.PipeSecurity pipeSecurity, System.Security.Principal.WellKnownSidType sid, System.IO.Pipes.PipeAccessRights pipeAccessRights) { }
+        public static System.IO.Pipes.PipeSecurity LocalOnly(this System.IO.Pipes.PipeSecurity pipeSecurity) { }
+        public static bool PipeExists(string pipeName, int timeout = 1) { }
+    }
+    public interface ISerializer
+    {
+        object? Deserialize(string json, System.Type type);
+        System.Threading.Tasks.ValueTask<T?> DeserializeAsync<T>(System.IO.Stream json, Microsoft.Extensions.Logging.ILogger? logger);
+        string Serialize(object? obj);
+        void Serialize(object? obj, System.IO.Stream stream);
+    }
+    public sealed class IpcClient
+    {
+        [System.Obsolete("Constructors of types with required members are not supported in this version of " +
+            "your compiler.", true)]
+        public IpcClient() { }
+        public UiPath.Ipc.ClientConfig Config { get; init; }
+        public UiPath.Ipc.ClientTransport Transport { get; init; }
+        public TProxy GetProxy<TProxy>()
+            where TProxy :  class { }
+    }
+    public class IpcProxy : System.Reflection.DispatchProxy, System.IDisposable
+    {
+        public IpcProxy() { }
+        public System.IO.Stream? Network { get; }
+        public event System.EventHandler ConnectionClosed;
+        public System.Threading.Tasks.ValueTask CloseConnection() { }
+        public void Dispose() { }
+        protected override object? Invoke(System.Reflection.MethodInfo? targetMethod, object?[]? args) { }
+    }
+    public sealed class IpcServer : System.IAsyncDisposable
+    {
+        [System.Obsolete("Constructors of types with required members are not supported in this version of " +
+            "your compiler.", true)]
+        public IpcServer() { }
+        public UiPath.Ipc.EndpointCollection Endpoints { get; init; }
+        public System.Collections.Generic.IReadOnlyList<UiPath.Ipc.ListenerConfig> Listeners { get; init; }
```

Comment:
I think we can improve the entity model to have better matching between client and server, and generally to simplify it.
The only real relationship differences I can see:
* A client has a single connection while a listener has many.  I don't see any problem with the API regarding this.
* A server can have multiple transport configurations "listeners". Of note, we could drop this feature of multiple listeners.  A user could simply create more servers calling the same configuration code.

I'd propose this model:
Endpoints = { Client, Server }
Endpoint = (IServiceProvider, Scheduler, RequestTimeout)
Client : Endpoint + (ClientTransport)
Server : Endpoint + (ServerTransport)